### PR TITLE
Mv 379 improper horizontal

### PR DESCRIPTION
**Pull request by pkrucz00**
The Pull Request aims to make the "Ooops, trn your device... " screen a little more accurate in terms of when it should pop out.

The proposed solution is that we check if a device is a smartphone (not a tablet - tablets should behave like desktops in terms of most UI) by:
   1. Finding out if it has a coarse pointer.
   2. Checking the width of a visual viewport:
        - if it's under MOBILE_WIDTH_BANDWIDTH then it is considered a smartphone
        - if not we check if the width is under `MAX_MOBILE_WIDTH_BANDWIDTH` and then check if he screen is horizontal.

The behaviour of the "Turn your device..." screen works as expected on most smartphones and tablets *tested on chrome devtools*. There may be some uncovered cases like a laptop with touchscreen and small viewport (eg. when minimising the width of a window). Those should be tested as well.

Besides, some small readability changes have been done..

### CODE REVIEW

**Review by bblaszkow06 — APPROVED** [review]
Looks fine to me, but wait for @kamil-stasiak's approval before merging.

File: assets/src/features/shared/hooks/useSmartphoneViewport.tsx
Comment (kamil-stasiak) [inline assets/src/features/shared/hooks/useSmartphoneViewport.tsx:41]: You could remove `useCallback` by placing functions inside `useEffect`

```suggestion
useEffect(() => {
    if (typeof window !== "undefined") {
      const updateIsSmartphoneState = () => {
        if (!window.visualViewport) return;

        const isCoarse = matchMedia("(pointer:coarse)").matches;
        const hasMobileWidth = window.visualViewport.width <= MOBILE_WIDTH_BREAKPOINT;

        const isLandscapedSmartphone =
          screen.orientation.type.includes("landscape") && window.visualViewport.width <= MAX_MOBILE_WIDTH_BREAKPOINT; // iPhone 12 max  viewpor width
        setIsSmartphone(isCoarse && (hasMobileWidth || isLandscapedSmartphone));
      }

      const updateIsHorizontalOrientationState = () => {
        if (!screen) return;

        setIsHorizontal(screen.orientation.type.includes("landscape"));
      }
      
      const updateState = () => {
        updateIsSmartphoneState();
        updateIsHorizontalOrientationState();
      }
      
      updateState();
      window.addEventListener("resize", updateState);
      return () => window.removeEventListener("resize", updateState);
    }
  }, []);
```

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]

**Review by kamil-stasiak — APPROVED** [review]

**Pull request merged**
